frontend/src/home/AutoSavePlan.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { debounce } from "lodash";
+import { batch, useDispatch, useSelector } from "react-redux";
+import { IPlanData } from "../models/types";
+import { updatePlanForUser } from "../services/PlanService";
+import { getAuthToken } from "../utils/auth-helpers";
+import {
+  setActivePlanStatusAction,
+  updateActivePlanAction,
+} from "../state/actions/userPlansActions";
+import { useCallback, useEffect } from "react";
+import { convertPlanToCreatePlanData } from "../utils/plan-helpers";
+import {
+  getActivePlanFromState,
+  getActivePlanStatusFromState,
+  getUserIdFromState,
+} from "../state";
+import { AppState } from "../state/reducers/state";
+import CheckCircleOutlineIcon from "@material-ui/icons/CheckCircleOutline";
+import styled from "styled-components";
+
+const Container = styled.div`
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+`;
+
+const DEBOUNCE_SAVE_DELAY_MS = 1000;
+
+export const AutoSavePlan: React.FC = () => {
+  const dispatch = useDispatch();
+  const { activePlanStatus, activePlan, userId } = useSelector(
+    (state: AppState) => ({
+      activePlanStatus: getActivePlanStatusFromState(state),
+      activePlan: getActivePlanFromState(state),
+      userId: getUserIdFromState(state),
+    })
+  );
+
+  const debouncedUpdate = useCallback(
+    debounce(async activePlan => {
+      await updatePlan(activePlan);
+    }, DEBOUNCE_SAVE_DELAY_MS),
+    []
+  );
+
+  useEffect(() => {
+    dispatch(setActivePlanStatusAction("Waiting to Update"));
+    debouncedUpdate(activePlan);
+  }, [
+    JSON.stringify(activePlan.schedule),
```

Comment:
A little worried that this might become slow because a schedule is a large object (although I did try it with a full example schedule and it seemed fine)

Reply:
Does the value need to be a string here for useEffect to work?

Reply:
It doesn't have to be a string, but useEffect uses shallow equal, so for objects it's safest to stringify them